src/MergeContext.js

```diff
@@ -541,24 +556,27 @@ class PullRequest {
         return Approval.GrantAfterTimeout("waiting for more votes or a slow burner timeout", Config.votingDelayMax() - prAgeMs);
     }
 
-    async _setApprovalStatus(sha) {
-        assert(sha);
-
+    async _setApprovalStatuses() {
         if (!Config.manageApprovalStatus())
             return;
 
-        const combinedStatus = await GH.getStatuses(sha);
-        const approvalStatus = combinedStatus.statuses ?
-            combinedStatus.statuses.find(el => el.context.trim() === Config.approvalContext()) : null;
-
-        if (approvalStatus && this._approval.matchesGitHubStatusCheck(approvalStatus)) {
-            this._log("approval status already exists: " + Config.approvalContext() + ", " + this._approval);
+        if (this._dryRun("setting approval status"))
```

Comment:
It looks like this check should be moved into _createApprovalStatus() because hasApprovalStatus() and addApprovalStatus() calls do not modify GitHub (and do allow us to get deeper in dry-run mode).